src/helpers/passwordGenerator.js

```diff
@@ -0,0 +1,52 @@
+export const passwordGenerator = (options) => {
+    // This method should not be used to create safe/permament passwords for users
+    // Passwords should be generated by browsers or password managers
+    // Only justifiable use-case is creating temporary passwords valid until first login
+    const uppercase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+    const lowercase = uppercase.toLowerCase();
+    const numbers = '0123456789';
+    const specialCharacters = '!@#$%^&*()_+-=[]{}|;:,.<>?';
+    const length = options?.length ?? 10;
+    const isNumberRequired = options?.isNumberRequired ?? true;
+    const isLowerCaseRequired = options?.isLowerCaseRequired ?? true;
+    const isUpperCaseRequired = options?.isUpperCaseRequired ?? true;
+    const isSpecialSymbolRequired = options?.isSpecialSymbolRequired ?? true;
+
+    let password  = "";
+
+    const getCategoriesArray = () => {
+        const availableCategories = []
+        if (isLowerCaseRequired){
+            availableCategories.push(lowercase); 
+        }
+        if(isUpperCaseRequired){
+            availableCategories.push(uppercase)
+        }
+        if(isNumberRequired){
+            availableCategories.push(numbers)
+        }
+        if(isSpecialSymbolRequired){
+            availableCategories.push(specialCharacters)
+        }
```

Comment:
If 'if' statement has just a one line, I would suggest to get rid of curly braces and do something like this:
`
  if (isLowerCaseRequired) availableCategories.push(lowercase);
  if (isUpperCaseRequired) availableCategories.push(uppercase);
  if (isNumberRequired) availableCategories.push(numbers);
  if (isSpecialSymbolRequired) availableCategories.push(specialCharacters);
`
This will reduce this block to 4 lines of code.